### PR TITLE
Refactor/server env config

### PR DESCRIPTION
**Pull request by AllMikeNoIke**

### CODE REVIEW

File: server/src/index.ts [hunk 1]
```diff
@@ -47,20 +54,20 @@ app.use(
     saveUninitialized: false,
     proxy: true,
     cookie: {
-      secure: IS_PROD, // SSL only in production
+      secure: process.env.NODE_ENV === "production", // SSL only in production
```
Comment (amirfakhrullah) [inline server/src/index.ts:57]: this is a bad pattern. I have made `IS_PROD` variable which already equals to `process.env.NODE_ENV === "production"`.  Putting in a variable is the better practice

File: server/src/index.ts [hunk 2]
```diff
@@ -21,20 +20,28 @@ import courseRouter from "./routes/courses.route";
 // instantiate express app
 const app = express();
 
+// mongodb connection
```
Comment (amirfakhrullah) [inline server/src/index.ts:23]: this should be the last one after we set up the routers. Please revert back

File: server/src/index.ts [hunk 3]
```diff
@@ -47,20 +54,20 @@ app.use(
...
-  })
+  }),
 );
 app.use(passport.initialize());
 app.use(passport.session());
 
 // Document API with Swagger if not in production
 // Docs available at /api/docs
-if (!IS_PROD) {
+if (process.env.NODE_ENV !== "production") {
```
Comment (amirfakhrullah) [inline server/src/index.ts:70]: please revert

File: server/package.json [hunk 4]
```diff
@@ -5,8 +5,8 @@
   "license": "MIT",
   "scripts": {
     "build": "tsc --project ./",
-    "start:dev": "nodemon src/index.ts",
-    "start": "node dist/index.js",
+    "start:dev": "NODE_ENV=development nodemon src/index.ts",
+    "start": "NODE_ENV=production node dist/index.js",
```
Comment (amirfakhrullah) [inline server/package.json:9]: running locally, NODE_ENV by default is already "development", and in heroku it's already "production". Please remove this one

**Comment by amirfakhrullah** [comment]
This PR will be closed as this is not a good refactoring

File: server/src/index.ts (hunk 1 above)
Comment (AllMikeNoIke) [reply server/src/index.ts:57]: thanks for the education!

**Pull request closed without merging**